### PR TITLE
Add a system property to ignore version of deserialized objects (models)

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

File: h2o-core/src/main/java/water/AutoBuffer.java [hunk 1]
```diff
@@ -281,15 +285,23 @@ public AutoBuffer( InputStream is ) {
...
+    checkVersion(getStr());
     String[] typeMap = getAStr();
     _typeMap = new short[typeMap.length];
     for( int i=0; i<typeMap.length; i++ )
       _typeMap[i] = (short)(typeMap[i]==null ? 0 : TypeMap.onIce(typeMap[i]));
   }
 
+  private void checkVersion(String version) {
+    final boolean ignoreVersion = Boolean.getBoolean(H2O_SYSTEM_SERIALIZATION_IGNORE_VERSION);
+    if (! version.equals(H2O.ABV.projectVersion())) {
```
Comment (jakubhava) [inline h2o-core/src/main/java/water/AutoBuffer.java]: extra space after `!` ?

File: h2o-core/src/main/java/water/AutoBuffer.java [hunk 2]
```diff
@@ -281,15 +285,23 @@ public AutoBuffer( InputStream is ) {
...
+      String msg = "Found version "+version+", but running version "+H2O.ABV.projectVersion();
```
Comment (jakubhava) [inline h2o-core/src/main/java/water/AutoBuffer.java]: maybe spaces around `+`

Comment (michalkurka) [reply h2o-core/src/main/java/water/AutoBuffer.java]: done

File: h2o-core/src/main/java/water/AutoBuffer.java (hunk 1 above)
Comment (michalkurka) [reply h2o-core/src/main/java/water/AutoBuffer.java]: I like it this way ;)

Comment (jakubhava) [reply h2o-core/src/main/java/water/AutoBuffer.java]: oki :)

**Pull request merged**